### PR TITLE
[SC] Refactored unused parameters and variables

### PR DESCRIPTION
**Pull request by hyunsooda**
## Proposed changes

Requested from #1260 

- Added context error handle in `local_backend.go`
- Removed duplicated context in `remote_backend.go`

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by kjhman21 — DISMISSED** [review]
LGTM. By the way, @hyunsooda Could you please check we have enough test cases for SC?

**Comment by hyunsooda** [comment]
@kjhman21  Do you want me to check that all the test cases were well set up for the entire code base in SC?

**Comment by kjhman21** [comment]
@hyunsooda Yeah, just skim through SC test cases, and see if it is enough or not.

**Comment by hyunsooda** [comment]
@kjhman21 It will be expected to take a long time as the bundle is not small. Let me post an issue or PR whenever I found improper stuff.

**Comment by kjhman21** [comment]
@henry-will @aidan-kwon @KimKyungup This one is quite simple one. Please take a look and approve it if there is no problem.

**Comment by kjhman21** [comment]
@henry-will @aidan-kwon @KimKyungup This one is quite simple one. Please take a look and approve it if there is no problem.

File: node/sc/local_backend.go [hunk 1]
```diff
@@ -202,16 +207,16 @@ func (lb *LocalBackend) EstimateGas(ctx context.Context, call klaytn.CallMsg) (g
...
 }
 
 // ChainID can return the chain ID of the chain.
-func (lb *LocalBackend) ChainID(ctx context.Context) (*big.Int, error) {
+func (lb *LocalBackend) ChainID(_dummyCtx context.Context) (*big.Int, error) {
 	return lb.config.ChainID, nil
 }
 
-func (lb *LocalBackend) TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error) {
+func (lb *LocalBackend) TransactionReceipt(txHash common.Hash) (*types.Receipt, error) {
```
Comment (aidan-kwon) [inline node/sc/local_backend.go:219]: Why did you remove ctx here?

File: node/sc/local_backend.go [hunk 2]
```diff
@@ -76,22 +76,27 @@ func (lb *LocalBackend) BalanceAt(ctx context.Context, account common.Address, b
...
+func (lb *LocalBackend) CallContract(_dummyCtx context.Context, call klaytn.CallMsg, blockNumber *big.Int) ([]byte, error) {
 	if blockNumber != nil && blockNumber.Cmp(lb.subbrige.blockchain.CurrentBlock().Number()) != 0 {
 		return nil, errBlockNumberUnsupported
 	}
 	currentState, err := lb.subbrige.blockchain.State()
 	if err != nil {
 		return nil, err
 	}
-	rval, _, _, err := lb.callContract(ctx, call, lb.subbrige.blockchain.CurrentBlock(), currentState)
+	rval, _, _, err := lb.callContract(call, lb.subbrige.blockchain.CurrentBlock(), currentState)
```
Comment (aidan-kwon) [inline node/sc/local_backend.go:95]: Is it okay that we neglect caller's context and generate a new context inside of `callContract`?

File: node/sc/local_backend.go [hunk 3]
```diff
@@ -152,11 +157,11 @@ func (lb *LocalBackend) PendingNonceAt(ctx context.Context, account common.Addre
 	return lb.subbrige.txPool.GetPendingNonce(account), nil
 }
 
-func (lb *LocalBackend) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
+func (lb *LocalBackend) SuggestGasPrice(_dummyCtx context.Context) (*big.Int, error) {
```
Comment (aidan-kwon) [inline node/sc/local_backend.go:160]: Why did you change `ctx` to `_dummyCtx` only in a few methods? It seems that other methods such as `PendingNonceAt` doesn't use `ctx` but you didn't change the name of it.

Comment (henry-will) [reply node/sc/local_backend.go:160]: _dummyCtx name is not good if it remains in codes, because other readers are confused, they don't know why it is used. When you want to rename unused parameters, it is good to remove as possible.

**Comment by hyunsooda** [comment]
The description of this PR's change was rewritten and please take a look again.

**Review by KimKyungup — APPROVED** [review]

**Review by hqjang-pepper — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]
LGTM
